### ISSUE TITLE
avahi-autoipd not fast enough to pass BCT tests

### ISSUE THREAD

**Issue by burakcayir** [lathiat/avahi#89]
Running avahi 0.6.32 for Bonjour Conformance tests and getting failures for Cable Change Handling test.

The operator disconnects and reconnects the device.  The test tool verifies that the device repeats its address probing to verify that its address is still unique.  The test tool denies the first probe and verifies that the device picks a new address and probes/announces again. 

So once we reassociate the device with network, avahi-autoipd goes to starting state very late, it claims the ip address only after the BCT test ends, So it could never catch this conflicting IP and try to pick a new one.
Note that if i run it as a daemon in the background, it never catches the conflict.

I had to manually kill the autoipd daemon and start it manually before the test started. In that case it passes the test.

Same problem occurs in initial probing too, autoipd does not go to probing state unless i reboot the device or reconnect it to the network.

Do you have any suggestions or a fix for short term in order to fully support BCT? Is there any successful combination of avai and autoipd configuration to make it pass all tests?

**Comment by lathiat** [comment]
Duplicate of #2 - basically we just haven't tried to make it pass any recent BCT.  PRs would be greatly appreciated if you have fixed this for your device.

**Comment by awy** [comment]
This is not really a duplicate of #2 although it is another issue related to Bonjour Conformance tests.

I have found with the latest BTCs, and avahi 0.7, that this test is passed.

Do you leave avahi-autoipd running while the network is disconnected? In my case it is stopped, and restarted when the network link comes back up, after first trying DHCP. This is all within any tester timeouts.
